posthog/api/feature_flag.py

```diff
@@ -319,7 +319,7 @@ def evaluation_reasons(self, request: request.Request, **kwargs):
         if not distinct_id:
             raise exceptions.ValidationError(detail="distinct_id is required")
 
-        flags, reasons = get_all_feature_flags(self.team_id, distinct_id, groups)
+        flags, reasons, _ = get_all_feature_flags(self.team_id, distinct_id, groups)
```

Comment:
We should add validation for `payloads` in this file, the 'key' can only be `true`  for non-multivariates, or match a variant name.

Reply:
(and test for APi sending invalid payload)

Reply:
done